src/main/java/com/SpringBoot/RestApi/UserManagement/UserController.java

```
@@ -0,0 +1,138 @@
+		
+		UserOtpRequests otpId=new UserOtpRequests();
+		
+		otpId.setUserId(createdUser.getUserId());
+		
+		otpId.setOtp(otp);
+		
+		otpId.setCreatedAt(timenow);
+		
+		otpId.setValidTill(validTill);
+		
+		otpRepo.save(otpId);
+		
+	    return ResponseEntity.ok().body("for adding user with details otp is sent on mobile number and otp is:" +otp);
+	    
+	}
+	
+	@PostMapping ("/verify")
+	public ResponseEntity<Object> verifyCredentials (@RequestBody VerifyOtp details ){
+		
+		String mobileNo=details.getMobileNo();
+		
+		int otp=details.getOtp();
+		
+		Optional<UserDetails> userWithMobileNo=Optional.ofNullable(userRepo.findByMobileNumber(mobileNo));
+		
+		if(userWithMobileNo.isPresent()) {
+			
+			UserDetails userWithDetails=userWithMobileNo.get();
+			
+			long userId=userWithDetails.getUserId();
+			
+			boolean isDeleted=userWithDetails.getIsDeleted();
+			
+			UserOtpRequests otpDetailsWithId=otpRepo.findByUserId(userId);
+			
+			LocalDateTime currentTime=Helper.currentDateTime();
+			
+			LocalDateTime validTill=otpDetailsWithId.getValidTill();
+			
+			int comparisonResult=validTill.compareTo(currentTime);
+			
+			if(otp==otpDetailsWithId.getOtp() && comparisonResult>=0) {
+				
+				String token=Helper.randomTokenGenerator();
+				
+				LocalDateTime timeOfTokenCreated=Helper.currentDateTime();
+				
+				UserTokens tokenDetails = new UserTokens();
+				
+				tokenDetails.setUserId(userId);
+				
+				tokenDetails.setToken(token);
+				
+				tokenDetails.setCreatedAt(timeOfTokenCreated);
+				
+				tokenDetails.setUpdatedAt(timeOfTokenCreated);
+				
+				tokenDetails.setIsDeleted(isDeleted);
+				
+				tokenRepo.save(tokenDetails);
+				
+				return ResponseEntity.ok().body("user credentials verified , token is generated and token is:" + token);
+				
+			}
+			
+		}
+		
+		return ResponseEntity.ok().body("couldn't verify credentials , please input correct mobile number with otp within 10 minutes after otp is generated.");
```

we need to throw exception in case user is not present.